packages/client/src/views/RegisterForm/RegisterForm.tsx

```diff
@@ -1380,7 +1366,7 @@ export function replaceInitialValues(
     initialValue:
       isUndefined(sectionValues[field.name]) ||
       isNull(sectionValues[field.name])
-        ? getInitialValue(field, data || {}, userDetails)
```

Comment:
This was previously what trigged updates for inputs with `initialValueKey` configuration and dependency enabled. The issue really was that from here you cannot access Formik field values in where the new fetch button state is stored. Besides that, I think the logic should be inside FormFieldGenerator anyways.